pkg/odo/cli/component/create.go

```diff
@@ -265,6 +285,109 @@ func (co *CreateOptions) setResourceLimits() error {
 
 // Complete completes create args
 func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	if experimental.IsExperimentalModeEnabled() {
```

Comment:
just a thought should we make separate  function `func (co *CreateOptions) completeDevfile(...)` and do like this here
```
if experimental.IsExperimentalModeEnabled() {
completeDevfile(...)
}
```

Reply:
By design after we implement devfile support for odo, odo needs to support both devfile and s2i components, but devfile component has higher priority. 

So the experimental flag will be removed finally, currently we don't separate the generic `Complete`, `Validate` and `Run` functions for all devfile support for odo features (eg odo create, odo push, odo url). I think the reason is it would be better for refactoring once we remove the experimental flag.